Engine/TypeData.cs

```diff
@@ -348,8 +349,31 @@ public Type Load()
             catch (Exception ex)
             {
                 failedLoad = true;
-                log.Error("Unable to load type '{0}' from '{1}'. Reason: '{2}'.", Name, Assembly.Location,
-                    ex.Message);
+                var reason = ex.Message;
+                // Create a special-case error message when the load error is due to a dotnet runtime mismatch.
+                // FileName is a qualified assembly name, e.g. System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
+                if (ex is FileNotFoundException notFound && notFound.FileName.StartsWith("system.runtime", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Build the reason string. I don't know if it's possible for the assembly to not be strong-named,
+                    // but let's handle that just in case.
+                    var filename = notFound.FileName;
+                    var dependencyVersion = ".NET";
+                    var versionPrefix = "System.Runtime, Version=";
+                    if (filename.StartsWith(versionPrefix, StringComparison.OrdinalIgnoreCase))
+                    {
+                        var commaIndex = filename.IndexOf(',', versionPrefix.Length);
+                        if (commaIndex == -1)
+                        {
+                            commaIndex = filename.Length;
+                        }
+                        var versionString = filename.Substring(versionPrefix.Length, commaIndex - versionPrefix.Length);
+                        dependencyVersion += $" {versionString}";
```

Comment:
if commaIndex is -1, the versionString will be empty, right?1
```suggestion
                        if (commaIndex == -1)
                        {
                            // no comma so it ends with "Version=X.Y.Z.W"
                            commaIndex = filename.Length;
                        }
                        var versionString = filename.Substring(versionPrefix.Length, commaIndex - versionPrefix.Length);
                        dependencyVersion += $" {versionString}";
```

Reply:
No, if commaIndex is -1, then there was no trailing comma. In the line above we already checked that the line contains `Version=`. 

This is meant to handle the edge case where the AssemblyName was not strongnamed, though I'm not sure if that can even happen.